presto-tpch/src/test/java/com/facebook/presto/tpch/statistics/ColumnStatisticsRecorder.java

```diff
@@ -33,10 +42,49 @@ public ColumnStatisticsRecorder(TpchColumnType type)
     void record(Comparable<?> value)
     {
         if (value != null) {
-            nonNullValues.add(value);
+            final Hasher hasher = hashFunction.newHasher();
+
+            switch (type.getBase()) {
+                case IDENTIFIER:
+                    hasher.putLong((Long) value);
+                    break;
+                case INTEGER:
+                case DATE:
+                    hasher.putInt((Integer) value);
+                    break;
+                case DOUBLE:
+                    hasher.putDouble((Double) value);
+                    break;
+                case VARCHAR:
+                    hasher.putString((String) value, StandardCharsets.UTF_8);
+                    break;
+            }
+            hll.addRaw(hasher.hash().asLong());
+
+            minMaxValues.add(value);
+
+            if (type.getBase() == TpchColumnType.Base.VARCHAR) {
+                varCharSize += ((String) value).length();
+            }
         }
     }
 
+    /**
+     * Merge statistics from another {@link ColumnStatisticsRecorder} into the current object
+     *
+     * @param other
+     * @return
+     */
+    public ColumnStatisticsRecorder mergeWith(ColumnStatisticsRecorder other)
+    {
+        checkArgument(type.equals(other.type), "Merging incompatible column statistics");
```

Comment:
what scenario will trigger this sanity check?

Reply:
Say you're merging stats of a VARCHAR column and DOUBLE column, you're *probably* doing something wrong. It's a (weak) sanity check for such a scenario